### ISSUE TITLE
Compatibility with OpenCV4

### ISSUE THREAD

**Issue by stevethearkiv** [PRHLT/pageDistanceBasedContourGenerator#8]
Would it be possible to get this codebase working with OpenCV4?

It is a bit of a challenge to compile and have this software co-exist with others on a more modern ubuntu installation for example.

**Comment by imagine5am** [comment]
If this could be done, it would be great.
It would make this repository very useful.
I am on Ubuntu 18, and I have spent hours on OpenCV 2.4.9 and I still haven't figured it out.

**Comment by stevethearkiv** [comment]
> If this could be done, it would be great.
> It would make this repository very useful.
> I am on Ubuntu 18, and I have spent hours on OpenCV 2.4.9 and I still haven't figured it out.

I had to drop down to ubuntu 16 to get it to work. It is stated in the readme that its not tested with newer libraries.

**Comment by imagine5am** [comment]
Yes, I have read what have stated. Your repository is very useful to me. 
So, thank you :+1:

**Comment by mcroissant** [comment]
Are you actually intrested by a PR working under opencv4 ? you can check https://github.com/mcroissant/pageDistanceBasedContourGenerator/tree/opencv4 meanwhile, I just ahve a few doubts about some Makefile changes
